tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/paging/AmqpPagingTest.java

```diff
@@ -118,4 +128,49 @@ public void testPaging() throws Exception {
       connection.close();
    }
 
+
+   @Test(timeout = 60000)
+   public void testNotBlockOnGlobalMaxSize() throws Exception {
+      final int MSG_SIZE = 1000;
+      final StringBuilder builder = new StringBuilder();
+      for (int i = 0; i < MSG_SIZE; i++) {
+         builder.append('0');
+      }
+      final String data = builder.toString();
+      final int MSG_COUNT = 3_000;
+
+      // sending size to explod max size
+      server.getPagingManager().addSize((int) ((PagingManagerImpl) server.getPagingManager()).getMaxSize());
+      server.getPagingManager().addSize(100_000);
+
+      server.getAddressSettingsRepository().addMatch("blockedQueue", new AddressSettings().setAddressFullMessagePolicy(AddressFullMessagePolicy.BLOCK));
+
+      ConnectionFactory factory = CFUtil.createConnectionFactory("AMQP", "tcp://localhost:5672");
```

Comment:
Can we make this test more generic as its already using just jms api and test the other protocols switching the connection factories to test them for this problem as we do in some other tests.

Reply:
this is a problem exclusive to AMQP. in AMQP the credit is per link, while in openwork and core it's per address..

Do you mean to run the same test with other protocols? is that what you asked?

Reply:
Yes. Just to be sure.

Reply:
ok, will do later today.. thanks for the review